spec.html

```diff
@@ -7872,6 +7877,149 @@ <h1>Forward Progress</h1>
+      </emu-note>
+      <emu-note>
+        <p>Implementations are not obligated to empty WeakRefs for maximal sets of non-live objects.</p>
+        <p>If an implementation chooses a non-live set _S_ in which to empty WeakRefs, it must empty WeakRefs for all objects in _S_ simultaneously. In other words, an implementation must not empty a WeakRef pointing to an object _obj_ without emptying out other WeakRefs that, if not emptied, could result in an execution that observes the Object value of _obj_.</p>
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-weakref-host-hooks">
+      <h1>Host Hooks</h1>
+
+      <emu-clause id="sec-host-cleanup-finalization-registry" aoid=HostEnqueueFinalizationRegistryCleanupJob>
+        <h1>HostEnqueueFinalizationRegistryCleanupJob ( _finalizationRegistry_ )</h1>
+
+        <p>The abstract operation HostEnqueueFinalizationRegistryCleanupJob takes argument _finalizationRegistry_ (a FinalizationRegistry). HostEnqueueFinalizationRegistryCleanupJob is an implementation-defined abstract operation that is expected to call CleanupFinalizationRegistry(_finalizationRegistry_) at some point in the future, if possible. The host's responsibility is to make this call at a time which does not interrupt synchronous ECMAScript code execution.</p>
+      </emu-clause>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-clear-kept-objects" aoid=ClearKeptObjects>
+    <h1>ClearKeptObjects ( )</h1>
+    <p>The abstract operation ClearKeptObjects takes no arguments. ECMAScript implementations are expected to call ClearKeptObjects when a synchronous sequence of ECMAScript executions completes. The following steps are performed:</p>
+    <emu-alg>
+      1. Let _agentRecord_ be the surrounding agent's Agent Record.
+      1. Set _agentRecord_.[[KeptAlive]] to a new empty List.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-addtokeptobjects" aoid="AddToKeptObjects">
+    <h1>AddToKeptObjects ( _object_ )</h1>
+    <p>The abstract operation AddToKeptObjects takes argument _object_ (an Object). It performs the following steps when called:</p>
+    <emu-alg>
+      1. Let _agentRecord_ be the surrounding agent's Agent Record.
+      1. Append _object_ to _agentRecord_.[[KeptAlive]].
+    </emu-alg>
+    <emu-note>
+      When the abstract operation AddToKeptObjects is called with a target object reference, it adds the target to a list that will point strongly at the target until ClearKeptObjects is called.
+    </emu-note>
+  </emu-clause>
+
+  <emu-clause id="sec-cleanup-finalization-registry" aoid="CleanupFinalizationRegistry">
+    <h1>CleanupFinalizationRegistry ( _finalizationRegistry_ [ , _callback_ ] )</h1>
```

Comment:
As far as I can tell, nothing in this specification ever passes the `_callback_` argument to this abstract operation - the only caller is in the prose

> `HostEnqueueFinalizationRegistryCleanupJob` is an implementation-defined abstract operation that is expected to call `CleanupFinalizationRegistry(finalizationRegistry)` at some point in the future

which omits the second argument. Looking at the proposal history, it seems that the second argument is for use by `cleanupSome`, which has been split out of this proposal. As such, I think it makes sense to drop it from this PR and add it back once necessary, since that change is straightforward.